net/golioth/settings.c

```diff
@@ -192,6 +214,8 @@ static int on_setting(const struct coap_packet *response,
 
 			if (setting_status != GOLIOTH_SETTINGS_SUCCESS) {
 				cumulative_status = setting_status;
+			} else {
+				save_setting(key, &value);
```

Comment:
I would change this to:
```
if (setting_status == GOLIOTH_SETTINGS_SUCCESS) {
	save_setting(key, &value);
} else {
	cumulative_status = setting_status;	
}
```
so that our mind doesn't have to negate twice (`!=` followed by `else`). It should improve readability.